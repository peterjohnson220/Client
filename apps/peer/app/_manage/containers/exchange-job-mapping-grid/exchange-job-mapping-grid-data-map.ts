export const associationStatus = [
    {
        Key: "Associated",
        Value: true
    },
    {
        Key: "Not Associated",
        Value: false
    }
]

export const associationPending = [
    {
        Key: "Pending Approval",
        Value: true
    }, {
        Key: "Active",
        Value: false
    }
]