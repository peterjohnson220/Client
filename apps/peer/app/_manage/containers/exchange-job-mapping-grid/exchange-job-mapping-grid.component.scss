@import 'kendo-grid';

.mapped .badge{
  width: 75px;
}

:host ::ng-deep .k-grid {
  tr, tr.k-alt {
    &:hover {
      button {
        visibility: visible;
      }
    }
  }
}

.match-button {
  visibility: hidden;
}
