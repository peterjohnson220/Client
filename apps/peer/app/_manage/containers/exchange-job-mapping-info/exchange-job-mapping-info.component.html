<div class="mapping-info-container card h-100">
  <div class="card-header">
    <div class="align-self-center mr-auto">
      Job Information
    </div>
    <ng-container *ngIf="selectedExchangeJobMapping.Mapped">
      <button id="add-mapping-button"
              class="btn btn-sm"
              [disabled]="selectedExchangeJobMapping.CompanyJobMappings.length > 9"
              [ngClass]="{ 'btn-secondary': !addingMapping, 'btn-warning': addingMapping }"
              (click)="toggleAdding()">
        {{ addingMapping ? 'Cancel' : 'Add Association' }}
      </button>
    </ng-container>

    <div (click)="close()"
         class="close-btn">
      <fa-icon [icon]="['far','times']"></fa-icon>
    </div>
  </div>
  <div class="card-body d-flex">
    <pf-job-info-container [job]="exchangeJobInfo">
    </pf-job-info-container>

    <div class="vertical-separator"></div>
    <pf-async-container class="company-job-mapping-container"
                        [opacityLevel]="0.4"
                        [noOpactity]="true"
                        [loading]="(applyingMapping$ | async) || (deletingMapping$ | async)  || (loadingExchangeMappingInfo$ | async)">
      <div class="company-job-mapping d-flex flex-column justify-content-between"
           *ngIf="selectedExchangeJobMapping.Mapped && !addingMapping ">
        <ngb-carousel [activeId]="getActiveCompanyJobId$ | async"
                      [interval]="0"
                      (slide)="onCarouselSlideChange($event)">
          <ng-container *ngFor="let mapping of selectedJobs; trackBy: trackById">
            <ng-template ngbSlide
                         [id]="mapping.CompanyJobId">
              <button class="delete-mapping-button btn btn-sm btn-danger"
                      [disabled]="(deletingMapping$ | async) || addingMapping"
                      (click)="handleDeleteClick()">
                <fa-icon icon="trash-alt" size="lg"></fa-icon>
              </button>
              <pf-job-info-container [job]="mapping"
                                     [showJobCode]="true">
              </pf-job-info-container>
            </ng-template>
          </ng-container>
        </ngb-carousel>
        <div class="carousel-controls d-flex flex-row justify-content-between"
             *ngIf="selectedJobs.length > 1">
          <div class="control control-left"
               (click)="handleControlLeftClick()">
            <fa-icon icon="chevron-circle-left" size="2x"></fa-icon>
          </div>
          <div class="indicators flex-grow-1 space-out-children text-center">
            <span class="control"
                  [ngClass]="{'active' : (getActiveCompanyJobId$ | async) ===  mapping.CompanyJobId}"
                  *ngFor="let mapping of (selectedCompanyJobInfoModels$ | async)"
                  (click)="handleIndicatorClick(mapping.CompanyJobId)">
              <fa-icon icon="circle"></fa-icon>
            </span>
          </div>
          <div class="control control-right"
               (click)="handleControlRightClick()">
            <fa-icon icon="chevron-circle-right" size="2x"></fa-icon>
          </div>
        </div>
      </div>

      <div class="company-job-mapping-add"
           *ngIf="!selectedExchangeJobMapping.Mapped || addingMapping">
        <div style="display:flex;">
          <pf-input-debounce [placeholderText]="'Search for a Job Code or Title...'"
                             [(ngModel)]="companyJobQuery"
                             [disabled]="applyingMapping$ | async"
                             (valueChanged)="handleSearchTitleValueChanged($event)"
                             class="search-box">
          </pf-input-debounce>
          <pf-input-debounce [placeholderText]="'Search in the Job Description...'"
                             [(ngModel)]="companyDescriptionQuery"
                             [disabled]="applyingMapping$ | async"
                             (valueChanged)="handleSearchDescValueChanged($event)"
                             class="search-box">
          </pf-input-debounce>
        </div>

        <div class="mapping-results-container"
             [class.remove-scroll]="(companyJobsToMapToLoading$ | async) || (loadingExchangeMappingInfo$ | async)">
          <pf-async-container [loading]="(companyJobsToMapToLoading$ | async) || (loadingExchangeMappingInfo$ | async)"
                              [loadingError]="companyJobsToMapToLoadingError$ | async"
                              [hideReloadButton]="true">

            <ng-container *ngIf="(companyJobsToMapTo$ | async).length">

              <pf-company-job-map-result *ngFor="let companyJobToMapTo of companyJobsToMapTo$ | async"
                                         [companyJobToMapTo]=companyJobToMapTo
                                         [applyingMapping]="applyingMapping$ | async"
                                         [applyingMappingError]="applyingMappingError$ | async"
                                         [selectedMapping]="(selectedMappingCompanyJobId$ | async) == companyJobToMapTo.CompanyJobId"
                                         (applyMapping)="handleApplyMapping($event)"
                                         [highlightFilter]="companyDescriptionQuery">
              </pf-company-job-map-result>

            </ng-container>

            <ng-container *ngIf="!(companyJobsToMapTo$ | async).length && !(companyJobsToMapToLoading$ | async)">

              <div class="d-flex flex-column mt-5 align-items-center">
                <h6 class="mb-1"
                    innerHTML="{{buildNoResultsString()}}"></h6>
                <h5><small class="text-muted">Please change your filter criteria to search again.</small></h5>
              </div>

            </ng-container>

          </pf-async-container>

        </div>

      </div>

    </pf-async-container>

  </div>

</div>

<pf-delete-mapping-confirmation-modal [selectedExchangeJobTitle]="selectedExchangeJobMapping.ExchangeJobTitle"
                                      [jobTitle]="selectedCompanyJob?.JobTitle"
                                      [exchangeJobToCompanyJobId]="selectedExchangeJobToCompanyJobId">
</pf-delete-mapping-confirmation-modal>
