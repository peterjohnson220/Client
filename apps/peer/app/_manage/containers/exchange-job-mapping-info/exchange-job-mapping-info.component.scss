@import 'colors';

.card-header{
  height: 38px;
  min-height: 38px;
}

.card-body{
  padding-left: 0;
  padding-right: 0;
  padding-bottom: .25rem;
}

.mapping-info-container{
  position: absolute;
  height: 100%;
  width: 100%;

  .card-header {
    display: flex;
    padding: 0.5rem .5rem 0.5rem 1.25rem;
  }

  .card-body {
    //Firefox requires that flexbox has a min-height to calculate flex-item heights [JP]
    min-height: 0;
  }
}

.mapping-info-container .card-header{
  display: flex;
  padding: 0.5rem .5rem 0.5rem 1.25rem;
}

.mapping-info-container .card-header .btn{
  margin-left: 0.5rem;
}

.remove-scroll{
  overflow: hidden;
}

.company-job-mapping-container {
  flex: 0 0 50%;
  position: relative;

  & > * {
    height: 100%;
  }

  .company-job-mapping-add {
    padding: 0 1rem 0 1rem;
    display: flex;
    flex-direction: column;

    .mapping-results-container {
      margin-top: 0.5rem;
      position: relative;
      height: 100%;
      overflow-y: auto;
    }
  }
}

.close-btn{
  cursor: pointer;
  padding: .25rem 0.75rem;
  margin-left: 0.5rem;
  align-self: center;
}

.close-btn:hover{
  background-color: $selection-hover-background;
}

.delete-mapping-button {
  float: right;
  margin: 0 1rem 0 -4em;
  position: relative;
}

.carousel-controls {
  width: 100%;
  align-items: center;
  padding: .5rem 1rem .5rem 1rem;

  .control {
    color: $inactive-button-color;
    cursor: pointer;
    &:hover {
      color: $body-background-color;
    }
  }

  .control-left, .control-right {
    &:hover {
      color: $body-background-color;
    }
  }

  .indicators > .active {
    color: $body-background-color;
  }
}

:host ::ng-deep ngb-carousel {
  outline: none;
  display: flex !important;
  flex-direction: column;
  height: 100%;

  .carousel-inner {
    display: flex;
  }

  .carousel-item.active {
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  .carousel-control-prev, .carousel-control-next, .carousel-indicators {
    display: none;
  }
}
