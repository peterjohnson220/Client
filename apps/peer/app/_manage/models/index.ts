export * from './exchange-job-request-candidate.model';
