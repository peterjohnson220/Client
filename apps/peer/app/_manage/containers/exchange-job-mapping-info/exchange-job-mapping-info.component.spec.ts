import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';
import { By } from '@angular/platform-browser';

import { Store, combineReducers, StoreModule } from '@ngrx/store';
import { NgbSlide } from '@ng-bootstrap/ng-bootstrap';
import { NgbSlideEvent } from '@ng-bootstrap/ng-bootstrap/carousel/carousel';
import spyOn = jest.spyOn;

import * as fromRootState from 'libs/state/state';
import { generateMockCompanyJobMapping, generateMockExchangeJobMapping, UpsertExchangeJobMapRequest } from 'libs/models/peer';
import { generateMockLatestCompanyJob } from 'libs/models/common';

import * as fromExchangeJobMappingInfoActions from '../../actions/exchange-job-mapping-info.actions';
import * as fromExchangeJobMappingGridActions from '../../actions/exchange-job-mapping-grid.actions';
import * as fromPeerManagementReducer from '../../reducers';
import { ExchangeJobMappingInfoComponent } from './exchange-job-mapping-info.component';
import {GridDataResult} from '@progress/kendo-angular-grid';

describe('Peer - Exchange Job Mapping Info', () => {
  let fixture: ComponentFixture<ExchangeJobMappingInfoComponent>;
  let instance: ExchangeJobMappingInfoComponent;

  let store: Store<fromPeerManagementReducer.State>;
  const exchangeJobMappingMock = generateMockExchangeJobMapping();
  const mockCarousel: any = { next: jest.fn(), prev: jest.fn(), select: jest.fn() };

  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          peer_manage: combineReducers(fromPeerManagementReducer.reducers)
        }),
      ],
      declarations: [
        ExchangeJobMappingInfoComponent,
        NgbSlide
      ],
      // Shallow Testing
      schemas: [NO_ERRORS_SCHEMA]
    });

    store = TestBed.inject(Store);

    fixture = TestBed.createComponent(ExchangeJobMappingInfoComponent);
    instance = fixture.componentInstance;

    instance.carousel = mockCarousel;
    // Set up the store to already have a selected exchange job mapping
    store.dispatch(new fromExchangeJobMappingGridActions.SetActiveExchangeJob(exchangeJobMappingMock));
  });

  it('should emit a closeClicked event, when the close button is clicked', () => {
    // Spy on the emit method for the closeClicked EventEmitter
    spyOn(instance.closeClicked, 'emit');

    fixture.detectChanges();

    // Find the close button in the template and trigger a click
    const closeButton = fixture.debugElement.query(By.css('.close-panel'));
    closeButton.triggerEventHandler('click', null);

    expect(instance.closeClicked.emit).toHaveBeenCalled();
  });


  it('should set the selectedExchangeJobMapping to the selected mapping from the store', () => {
    fixture.detectChanges();

    expect(instance.selectedExchangeJobMapping).toBe(exchangeJobMappingMock);
  });

  it('should set the companyJobQuery to an empty string when a new exchange job mapping is selected', () => {
    fixture.detectChanges();

    instance.companyJobQuery = 'Blah Blah Blah';

    const newExchangeJobMappingSelection = { ...exchangeJobMappingMock, ExchangeJobId: 3983457 };

    store.dispatch(new fromExchangeJobMappingGridActions.SetActiveExchangeJob(newExchangeJobMappingSelection));

    expect(instance.companyJobQuery).toBe('');
  });

  it('should dispatch a LoadCompanyJobsToMapToByQuery action when a exchange job mapping that is not mapped is selected', () => {
    // Still need the actual implementation since we are making a setup call
    spyOn(store, 'dispatch');

    instance.exchangeId = 1;
    fixture.detectChanges();

    // Update store to have a selected mapping that is not mapped
    const newExchangeJobMappingSelection = { ...exchangeJobMappingMock, CompanyJobMappings: null };
    store.dispatch(new fromExchangeJobMappingGridActions.SetActiveExchangeJob(newExchangeJobMappingSelection));

    const expectedAction = new fromExchangeJobMappingInfoActions.LoadCompanyJobsToMapToByQuery({
      exchangeId: instance.exchangeId,
      jobTitleAndCodeQuery: exchangeJobMappingMock.ExchangeJobTitle,
      jobDescriptionQuery: ''
    });

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should build job models, when receiving a mapped company job', () => {
    fixture.detectChanges();

    const mappedCompanyJobs = generateMockLatestCompanyJob();

    store.dispatch(new fromExchangeJobMappingInfoActions.LoadMappedCompanyJobsSuccess([mappedCompanyJobs]));

    expect(instance.selectedJobs).toEqual([mappedCompanyJobs]);
  });

  it('should dispatch a LoadCompanyJobsToMapToByQuery action, when handling the search value changing', () => {
    instance.exchangeId = 1;
    instance.companyJobQuery = 'Accountant';
    instance.companyDescriptionQuery = 'Job Description';
    spyOn(store, 'dispatch');

    const payload = { exchangeId: instance.exchangeId, jobTitleAndCodeQuery: 'Accountant', jobDescriptionQuery: 'Job Description' };
    const expectedAction = new fromExchangeJobMappingInfoActions.LoadCompanyJobsToMapToByQuery(payload);
    instance.searchChanged();

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should dispatch a ApplyMapping action, when handling an apply mapping event', () => {
    instance.exchangeId = 1;
    fixture.detectChanges();
    spyOn(store, 'dispatch');

    const payload: UpsertExchangeJobMapRequest = {
      ExchangeId: instance.exchangeId,
      ExchangeJobId: exchangeJobMappingMock.ExchangeJobId,
      CompanyJobId: 234897
    };

    const expectedAction = new fromExchangeJobMappingInfoActions.ApplyMapping(payload);
    instance.handleApplyMapping(234897);

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should dispatch a CancelAddMapping action, when we are adding a mapping and cancel adding is clicked', () => {
    fixture.detectChanges();

    store.dispatch(new fromExchangeJobMappingInfoActions.AddMapping());

    spyOn(store, 'dispatch');

    // Find the cancel editing button in the template and trigger a click
    const cancelEditingBtn = fixture.debugElement.query(By.css('.card-header #add-mapping-button'));
    cancelEditingBtn.triggerEventHandler('click', null);

    const expectedAction = new fromExchangeJobMappingInfoActions.CancelAddMapping();

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should dispatch a AddMapping action, when clicking the add mapping button', () => {
    fixture.detectChanges();
    spyOn(store, 'dispatch');

    // Find the edit mapping button in the template and trigger a click
    const editMappingBtn = fixture.debugElement.query(By.css('.card-header #add-mapping-button'));
    editMappingBtn.triggerEventHandler('click', null);

    const expectedAction = new fromExchangeJobMappingInfoActions.AddMapping();

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should dispatch a LoadCompanyJobsToMapToByQuery action, when clicking the add mapping button', () => {
    instance.exchangeId = 1;
    fixture.detectChanges();
    spyOn(store, 'dispatch');

    // Find the edit mapping button in the template and trigger a click
    const editMappingBtn = fixture.debugElement.query(By.css('.card-header #add-mapping-button'));
    editMappingBtn.triggerEventHandler('click', null);

    const expectedAction = new fromExchangeJobMappingInfoActions.LoadCompanyJobsToMapToByQuery({
      exchangeId: instance.exchangeId,
      jobTitleAndCodeQuery: exchangeJobMappingMock.ExchangeJobTitle,
      jobDescriptionQuery: ''
    });
    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should display an add mapping button, when the selected exchange job mapping is mapped', () => {
    // Mock Exchange Job Mapping set up above is mapped already.
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should NOT display an add mapping button, when the selected exchange job mapping is not mapped', () => {
    const unmappedExchangeJobMock = { ...exchangeJobMappingMock, Mapped: false };

    store.dispatch(new fromExchangeJobMappingGridActions.SetActiveExchangeJob(unmappedExchangeJobMock));

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should display a carousel when more than one company job is mapped', () => {

    const exchangeJobMapping = generateMockExchangeJobMapping();
    const exchangeJobMappings = [{exchangeJobMapping}, {...exchangeJobMapping}];
    const grid: GridDataResult = {data: exchangeJobMappings, total: exchangeJobMappings.length};
    store.dispatch(new fromExchangeJobMappingGridActions.LoadExchangeJobMappingsSuccess(grid));
    fixture.detectChanges();
    store.dispatch(new fromExchangeJobMappingGridActions.SetActiveExchangeJob(exchangeJobMapping));
    fixture.detectChanges();
    const latestJob = generateMockLatestCompanyJob();
    const latestJobs = [latestJob, { ...latestJob, CompanyJobId: 234234 }];

    instance.selectedExchangeJobMapping = {...generateMockExchangeJobMapping(), Mapped: true};
    store.dispatch(new fromExchangeJobMappingInfoActions.LoadMappedCompanyJobsSuccess(latestJobs));

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it(`should NOT dispatch SetActiveMapping with the ExchangeJobToCompanyJobId of the first company job mapping on init
  when there are NO companyJobMappings`, () => {
      const mockCompanyJobMapping = generateMockCompanyJobMapping();
      const exchangeJobMappingWithMultipleMappingsMock = { ...exchangeJobMappingMock, CompanyJobMappings: [] };
      const expectedAction = new fromExchangeJobMappingInfoActions.SetActiveMapping(mockCompanyJobMapping.ExchangeJobToCompanyJobId);

      store.dispatch(new fromExchangeJobMappingGridActions.SetActiveExchangeJob(exchangeJobMappingWithMultipleMappingsMock));

      spyOn(store, 'dispatch');

      fixture.detectChanges();

      expect(store.dispatch).not.toBeCalledWith(expectedAction);
    });

  // TODO: Needs to be re-written, selectedCompanyJob does not match the mock type
  // it(`should NOT dispatch SetActiveMapping with the ExchangeJobToCompanyJobId of the first company job mapping on init
  // when the selectedCompanyJobMapping is no longer mapped`, () => {
  //     const mockCompanyJobMapping = generateMockCompanyJobMapping();
  //     const companyJobMappings = [mockCompanyJobMapping, { ...mockCompanyJobMapping, ExchangeJobToCompanyJobId: 2 }];
  //     const exchangeJobMappingWithMultipleMappingsMock = { ...exchangeJobMappingMock, CompanyJobMappings: companyJobMappings };
  //     const expectedAction = new fromExchangeJobMappingInfoActions.SetActiveMapping(mockCompanyJobMapping.ExchangeJobToCompanyJobId);
  //
  //     instance.selectedCompanyJob = mockCompanyJobMapping;
  //     store.dispatch(new fromExchangeJobMappingGridActions.SetActiveExchangeJob(exchangeJobMappingWithMultipleMappingsMock));
  //
  //     spyOn(store, 'dispatch');
  //
  //     fixture.detectChanges();
  //
  //     expect(store.dispatch).not.toBeCalledWith(expectedAction);
  //   });

  it(`should update selectedCompanyJob when the activeExchangeJobToCompanyJobId$ is changed`, () => {
    const mockCompanyJob = generateMockLatestCompanyJob();
    const companyJobMappings = [mockCompanyJob, { ...mockCompanyJob, CompanyJobId: 345 }];
    const expectedCompanyJobMapping = companyJobMappings[0];

    const mockExchangeJobMapping = generateMockExchangeJobMapping();

    store.dispatch(new fromExchangeJobMappingGridActions.SetActiveExchangeJob(mockExchangeJobMapping));

    fixture.detectChanges();

    store.dispatch((new fromExchangeJobMappingInfoActions.LoadMappedCompanyJobsSuccess(companyJobMappings)));

    fixture.detectChanges();

    expect(instance.selectedCompanyJob).toEqual(expectedCompanyJobMapping);
  });

  it(`should dispatch an OpenDeleteConfirmationModal action when handleDeleteClick is triggered`, () => {
    const expectedAction = new fromExchangeJobMappingInfoActions.OpenDeleteConfirmationModal();

    spyOn(store, 'dispatch');

    fixture.detectChanges();

    instance.handleDeleteClick();

    expect(store.dispatch).toBeCalledWith(expectedAction);
  });

  it(`should call carousel.next when handleControlRightClick is triggered`, () => {
    spyOn(instance.carousel, 'next');

    fixture.detectChanges();

    instance.carousel = mockCarousel;

    instance.handleControlRightClick();

    expect(instance.carousel.next).toBeCalled();
  });

  it(`should call carousel.prev when handleControlLeftClick is triggered`, () => {
    spyOn(instance.carousel, 'prev');

    fixture.detectChanges();

    instance.carousel = mockCarousel;

    instance.handleControlLeftClick();

    expect(instance.carousel.prev).toBeCalled();
  });

  it(`should call carousel.select when handleIndicatorClick is triggered`, () => {
    const expectedSlideId = exchangeJobMappingMock.CompanyJobMappings[0].ExchangeJobToCompanyJobId.toString();

    spyOn(instance.carousel, 'select');

    fixture.detectChanges();

    instance.carousel = mockCarousel;

    instance.handleIndicatorClick(expectedSlideId);

    expect(instance.carousel.select).toBeCalledWith(expectedSlideId);
  });

  it(`should dispatch a SetActiveMapping action with the slide event's current value when onCarouselSlideChange is triggered`, () => {
    const expectedSlideId = exchangeJobMappingMock.CompanyJobMappings[0].ExchangeJobToCompanyJobId;
    const expectedAction = new fromExchangeJobMappingInfoActions.SetActiveMapping(expectedSlideId);
    const mockSlideEvent: NgbSlideEvent = {
      paused: false,
      current: expectedSlideId.toString(),
      direction: null,
      prev: '0'
    };

    spyOn(store, 'dispatch');

    fixture.detectChanges();

    instance.onCarouselSlideChange(mockSlideEvent);

    expect(store.dispatch).toBeCalledWith(expectedAction);
  });
});
