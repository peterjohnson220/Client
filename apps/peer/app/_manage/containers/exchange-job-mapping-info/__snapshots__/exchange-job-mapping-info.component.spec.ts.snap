// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Peer - Exchange Job Mapping Info should NOT display an add mapping button, when the selected exchange job mapping is not mapped 1`] = `
<pf-exchange-job-mapping-info
  activeExchangeJobToCompanyJobIdSubscription={[Function Subscriber]}
  addingMapping="false"
  addingMapping$={[Function Store]}
  addingMappingSubscription={[Function Subscriber]}
  applyingMapping$={[Function Store]}
  applyingMappingError$={[Function Store]}
  carousel="undefined"
  closeClicked={[Function EventEmitter]}
  companyDescriptionQuery=""
  companyJobQuery=""
  companyJobsToMapTo$={[Function Store]}
  companyJobsToMapToLoading$={[Function Store]}
  companyJobsToMapToLoadingError$={[Function Store]}
  debouncedQueryValue=""
  deletingMapping$={[Function Store]}
  deletingMappingError$={[Function Store]}
  editMappingClicked={[Function EventEmitter]}
  exchangeJobInfo={[Function Object]}
  getActiveCompanyJobId$={[Function Store]}
  loadingExchangeMappingInfo$={[Function Store]}
  selectedCompanyJobInfoModels$={[Function Store]}
  selectedCompanyJobsSubscription={[Function Subscriber]}
  selectedExchangeJobMapping={[Function Object]}
  selectedExchangeJobMapping$={[Function Store]}
  selectedExchangeJobMappingSubscription={[Function Subscriber]}
  selectedJobs={[Function Array]}
  selectedMappingCompanyJobId$={[Function Store]}
  store={[Function Store]}
>
  <div
    class="mapping-info-container card h-100"
  >
    <div
      class="card-header"
    >
      <div
        class="align-self-center mr-auto"
      >
         Job Information 
      </div>
      
      <div
        class="close-btn"
      >
        <fa-icon />
      </div>
    </div>
    <div
      class="card-body d-flex"
    >
      <pf-job-info-container />
      <div
        class="vertical-separator"
      />
      <pf-async-container
        class="company-job-mapping-container"
      >
        
        
        <div
          class="company-job-mapping-add"
        >
          <div
            style="display:flex;"
          >
            <pf-input-debounce
              class="search-box"
            />
            <pf-input-debounce
              class="search-box"
            />
          </div>
          <div
            class="mapping-results-container remove-scroll"
          >
            <pf-async-container>
              
              
            </pf-async-container>
          </div>
        </div>
      </pf-async-container>
    </div>
  </div><pf-delete-mapping-confirmation-modal />
</pf-exchange-job-mapping-info>
`;

exports[`Peer - Exchange Job Mapping Info should display a carousel when more than one company job is mapped 1`] = `
<pf-exchange-job-mapping-info
  activeExchangeJobToCompanyJobIdSubscription={[Function Subscriber]}
  addingMapping="false"
  addingMapping$={[Function Store]}
  addingMappingSubscription={[Function Subscriber]}
  applyingMapping$={[Function Store]}
  applyingMappingError$={[Function Store]}
  carousel="undefined"
  closeClicked={[Function EventEmitter]}
  companyDescriptionQuery=""
  companyJobQuery=""
  companyJobsToMapTo$={[Function Store]}
  companyJobsToMapToLoading$={[Function Store]}
  companyJobsToMapToLoadingError$={[Function Store]}
  debouncedQueryValue=""
  deletingMapping$={[Function Store]}
  deletingMappingError$={[Function Store]}
  editMappingClicked={[Function EventEmitter]}
  exchangeJobInfo={[Function Object]}
  getActiveCompanyJobId$={[Function Store]}
  loadingExchangeMappingInfo$={[Function Store]}
  selectedCompanyJob={[Function Object]}
  selectedCompanyJobInfoModels$={[Function Store]}
  selectedCompanyJobsSubscription={[Function Subscriber]}
  selectedExchangeJobMapping={[Function Object]}
  selectedExchangeJobMapping$={[Function Store]}
  selectedExchangeJobMappingSubscription={[Function Subscriber]}
  selectedJobs={[Function Array]}
  selectedMappingCompanyJobId$={[Function Store]}
  store={[Function Store]}
>
  <div
    class="mapping-info-container card h-100"
  >
    <div
      class="card-header"
    >
      <div
        class="align-self-center mr-auto"
      >
         Job Information 
      </div>
      
      
      <button
        class="btn btn-sm btn-secondary"
        id="add-mapping-button"
        ng-reflect-klass="btn btn-sm"
        ng-reflect-ng-class="[object Object]"
      >
         Add Association 
      </button>
      <div
        class="close-btn"
      >
        <fa-icon />
      </div>
    </div>
    <div
      class="card-body d-flex"
    >
      <pf-job-info-container />
      <div
        class="vertical-separator"
      />
      <pf-async-container
        class="company-job-mapping-container"
      >
        
        <div
          class="company-job-mapping d-flex flex-column justify-content-between"
        >
          <ngb-carousel>
            
            
            
            
            
          </ngb-carousel>
          
          <div
            class="carousel-controls d-flex flex-row justify-content-between"
          >
            <div
              class="control control-left"
            >
              <fa-icon
                icon="chevron-circle-left"
                size="2x"
              />
            </div>
            <div
              class="indicators flex-grow-1 space-out-children text-center"
            >
              
              <span
                class="control active"
                ng-reflect-klass="control"
                ng-reflect-ng-class="[object Object]"
              >
                <fa-icon
                  icon="circle"
                />
              </span>
              <span
                class="control"
                ng-reflect-klass="control"
                ng-reflect-ng-class="[object Object]"
              >
                <fa-icon
                  icon="circle"
                />
              </span>
            </div>
            <div
              class="control control-right"
            >
              <fa-icon
                icon="chevron-circle-right"
                size="2x"
              />
            </div>
          </div>
        </div>
        
      </pf-async-container>
    </div>
  </div><pf-delete-mapping-confirmation-modal />
</pf-exchange-job-mapping-info>
`;

exports[`Peer - Exchange Job Mapping Info should display an add mapping button, when the selected exchange job mapping is mapped 1`] = `
<pf-exchange-job-mapping-info
  activeExchangeJobToCompanyJobIdSubscription={[Function Subscriber]}
  addingMapping="false"
  addingMapping$={[Function Store]}
  addingMappingSubscription={[Function Subscriber]}
  applyingMapping$={[Function Store]}
  applyingMappingError$={[Function Store]}
  carousel="undefined"
  closeClicked={[Function EventEmitter]}
  companyDescriptionQuery=""
  companyJobQuery=""
  companyJobsToMapTo$={[Function Store]}
  companyJobsToMapToLoading$={[Function Store]}
  companyJobsToMapToLoadingError$={[Function Store]}
  debouncedQueryValue=""
  deletingMapping$={[Function Store]}
  deletingMappingError$={[Function Store]}
  editMappingClicked={[Function EventEmitter]}
  exchangeJobInfo={[Function Object]}
  getActiveCompanyJobId$={[Function Store]}
  loadingExchangeMappingInfo$={[Function Store]}
  selectedCompanyJobInfoModels$={[Function Store]}
  selectedCompanyJobsSubscription={[Function Subscriber]}
  selectedExchangeJobMapping={[Function Object]}
  selectedExchangeJobMapping$={[Function Store]}
  selectedExchangeJobMappingSubscription={[Function Subscriber]}
  selectedJobs={[Function Array]}
  selectedMappingCompanyJobId$={[Function Store]}
  store={[Function Store]}
>
  <div
    class="mapping-info-container card h-100"
  >
    <div
      class="card-header"
    >
      <div
        class="align-self-center mr-auto"
      >
         Job Information 
      </div>
      
      
      <button
        class="btn btn-sm btn-secondary"
        id="add-mapping-button"
        ng-reflect-klass="btn btn-sm"
        ng-reflect-ng-class="[object Object]"
      >
         Add Association 
      </button>
      <div
        class="close-btn"
      >
        <fa-icon />
      </div>
    </div>
    <div
      class="card-body d-flex"
    >
      <pf-job-info-container />
      <div
        class="vertical-separator"
      />
      <pf-async-container
        class="company-job-mapping-container"
      >
        
        <div
          class="company-job-mapping d-flex flex-column justify-content-between"
        >
          <ngb-carousel>
            
          </ngb-carousel>
          
        </div>
        
      </pf-async-container>
    </div>
  </div><pf-delete-mapping-confirmation-modal />
</pf-exchange-job-mapping-info>
`;
